test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -261,6 +270,8 @@ private static void validateSpec(RestSpec restSpec) {
 
     @After
     public void wipeCluster() throws Exception {
+        // Running tasks might create indices and otherwise invalidate the wipe. Lets just let the finish.
+        waitForAllRunningTasksToFinish();
```

Comment:
> I'm still confused. How did this cause tests to fail then?

Another test tried to create an index with the same name.

> Also, why are we kicking off a task in a rest test but never waiting for it to complete? Seems like a reindex rest test succeeding should only happen if the task completed successfully?

There is no "please only return when this task has successfully completed" API. It'd be a great thing to have one day! For now if you explicitly set `wait_for_completion=false` then it us up to the user to monitor the task. And we have a rest test that verifies that setting wait_for_completion=false returns the task id so the user can do just that. But it can't wait for the task to finish because we don't have that API.

We should build that API. And we will. But we don't have time to build it right now.

Another idea:

Instead of waiting for all tasks to finish after each REST test we just fail the test if there are any tasks still running. And we add a new section that we can use to "wait for the task I just started to finish". When we have the API we can replace the section with the API.